### DISCUSSION TITLE
Charging speed depending on battery level

### DISCUSSION THREAD

**Discussion by elhennig** [VR-25/acc#192] [General]
Is there an option the charge the battery only slowly after reaching a certain battery level. I only found how to set the charging current in general and limit it to a certain value.
I have a scenario in mind to charge the battery until 75% rapidly if possible and the limit the current to a rather low value like 300-500 mA. Is that already doable?

**Comment by VR-25** [comment]
While there's no dedicated mechanism for that, it's fairly simple to set up.
Share your config.txt file after setting your regular current limit.

**Comment by elhennig** [comment]
This is the config I use.
[acc_config.txt](https://github.com/VR-25/acc/files/12138580/acc_config.txt)

**Comment by VR-25** [comment]
I was expecting to see a current limit setting there.

Anyway...
After setting your regular limit, e.g., `acc -sc 2000` (2000 mA), find the line `maxChargingCurrent=(...)` in your config and add the following line below it:

`: limit current to 500 mA at 75%; [ $(cat ${battCapacity:-//NULL} 2>/dev/null || echo 75) -lt 75 ] || maxChargingCurrent=(YOUR CANGES HERE)`

Essentially, you're going to replace all instances of 2000 with 500.

**Comment by elhennig** [comment]
I did not impose any limit, yet. Sorry. This is what I used different chargers for ;-)

Anyway you config seems to work, at least most of the times, sometimes the values seem to be higher than the configured value

**Comment by VR-25** [comment]
It's a kernel/hardware limitation. Some values/devices work better than others.
